### DISCUSSION TITLE
Inner class resources not saving (v4.2.beta1)

### DISCUSSION THREAD

**Discussion by donn-xx** [godotengine/godot-proposals#8178] [Scripting]
Version: v4.2.beta1.official [b1371806a]

Not sure if this is a bug or my own misunderstanding. I made a small project to show the issue:
https://gitlab.com/dbat/inner_class_resource_issue

Open issue.tscn and check the code.

### The story
I tried to use an inner class like this:
```
class Stuff extends Resource:
	@export var id:String
	@export var position:Vector2
	@export var resource:Resource
```
That worked! I was happy. Then I extended things so I could instance Stuff and put that instance into a Dictionary. The idea is that my store.gd script can hold a bunch of 'stuff'.

I then put that into a test scene to see how it behaved. I have another dict in my Node2D script called 'stores'. Pressing the Go checkbox (in the inspector) makes the store, fills it with some data and puts that into stores. Then it prints some data from all that to show it exists.

Then I saved the project and all resources and then closed Godot.

Then I opened Godot again and the data within stuff is missing.

### Screenshots
Before closing: Notice id, position and resource within "id001".
![before_restarting_godot](https://github.com/godotengine/godot-proposals/assets/392218/e9c53379-24b9-455c-8253-bc9d59fc2fae)
After reopening: It's all gone.
![after_restarting_godot](https://github.com/godotengine/godot-proposals/assets/392218/b7ff0fe8-a57e-4ba0-b745-78fd6085548d)

**Comment by Sslaxx** [comment]
Seems to be some confusion as to where bug reports go? This should be directly in the Godot Engine's issue tracker - https://github.com/godotengine/godot/issues - as this is an engine bug, not a proposal for enhancement or new issues.

**Reply by donn-xx** [reply]
Sorry, I was asking here in case it was not a bug or was known. I could not find anything in a search.

**Reply by Sslaxx** [reply]
If you're not sure if it's a bug or not, the main issue tracker is the place to ask.

**Reply by AThousandShips** [reply]
If you're not sure it's a bug the other community channels is the first place to ask, then a bug report 

This is for feature discussion

**Reply by donn-xx** [reply]
I'll gladly move it or repost it. Which other community channels? Can you give me a link?

**Reply by AThousandShips** [reply]
[These](https://godotengine.org/community)
